### PR TITLE
Add a bit more guidance to the autofac upgrade guide

### PR DESCRIPTION
**Pull request by danielmarbach**

### CODE REVIEW

File: nservicebus/upgrades/7to8/dependency-injection.md [hunk 1]
```diff
@@ -154,6 +154,21 @@ var autofacContainer = containerBuilder.Build();
...
+
+```csharp
+endpointConfiguration.UseContainer<AutofacBuilder>(
+    customizations: customizations =>
+    {
+        customizations.ExistingLifetimeScope(container);
+    });
+```
+
+it is recommended to either
```
Comment (andreasohlund) [inline nservicebus/upgrades/7to8/dependency-injection.md:167]:
```suggestion
it is recommended to either:
```

File: nservicebus/upgrades/7to8/dependency-injection.md [hunk 2]
```diff
@@ -154,6 +154,21 @@ var autofacContainer = containerBuilder.Build();
...
+
+- Stop building the container early and pass around the container builder to the various bootstrapping parts of the application _or_
```
Comment (andreasohlund) [inline nservicebus/upgrades/7to8/dependency-injection.md:169]:
```suggestion
- Avoid building the container early by passing the container builder to the various bootstrapping parts of the application _or_
```

File: nservicebus/upgrades/7to8/dependency-injection.md [hunk 3]
```diff
@@ -154,6 +154,21 @@ var autofacContainer = containerBuilder.Build();
...
+- Consider creating a [dedicated child lifetime scope](https://autofac.readthedocs.io/en/latest/integration/netcore.html#using-a-child-scope-as-a-root)
```
Comment (andreasohlund) [inline nservicebus/upgrades/7to8/dependency-injection.md:170]:
```suggestion
- Using a [dedicated child lifetime scope](https://autofac.readthedocs.io/en/latest/integration/netcore.html#using-a-child-scope-as-a-root)
```

Comment (andreasohlund) [reply nservicebus/upgrades/7to8/dependency-injection.md:170]: Nitpick: Perhaps end with what this achieves? ( Is it to "make sure that XYZ is the root container"?)

**Review by andreasohlund — APPROVED** [review]

File: nservicebus/upgrades/7to8/dependency-injection.md [hunk 4]
```diff
@@ -154,6 +154,23 @@ var autofacContainer = containerBuilder.Build();
...
+    customizations: customizations =>
+    {
+        customizations.ExistingLifetimeScope(container);
+    });
+```
+
+it is recommended to either:
+
+- Avoid building the container early by passing the container builder to the various bootstrapping parts of the application _or_
+- Using a [dedicated child lifetime scope](https://autofac.readthedocs.io/en/latest/integration/netcore.html#using-a-child-scope-as-a-root)
```
Comment (danielmarbach) [inline nservicebus/upgrades/7to8/dependency-injection.md:172]:
```suggestion
- Using a [dedicated child lifetime scope](https://autofac.readthedocs.io/en/latest/integration/netcore.html#using-a-child-scope-as-a-root) which allows accessing singletons of the root container while having dependencies managed by the service collection being "rooted" within the child scope. 
```

**Pull request merged**